[WIP] [DO NOT MERGE] Carousel: Use CSS Grid for fixed info footer, add icons

andrewserong opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

***Note: work on this change is progressing in #20145***

This is a WIP experiment, hacking around using CSS grid for laying out the info footer for the Jetpack Carousel. This exploration is branched from Ramon's work in https://github.com/Automattic/jetpack/pull/20108.

Possible fix for: Automattic/view-design#281 and resolves Automattic/view-design#284

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Add an apparent fixed footer to the Jetpack Carousel, with info and comment buttons
* Hide the extra info area beneath the footer by default, to prevent vertical scrolling
* When the info or comment button is clicked, reveal

#### Screenshot of WIP

![carousel-footer-reveal-with-pages-sml](https://user-images.githubusercontent.com/14988353/122881065-1402cd00-d37e-11eb-9ce4-184448b227e4.gif)

#### To-dos

* [x] Should we hide the extra footer area when navigating back / forward through slides?
* [ ] Add smooth scrolling to scrolling up / down
* [ ] Add a timeout before toggling the show class _off_ so that we can smooth scroll up _before_ the extra info content is hidden
* [ ] Tweak the tap targets of the buttons so that they are larger — currently (with my thumbs at least) it's difficult to tap the buttons on mobile (iPhone 8)
* [ ] Tweak styling and positioning of buttons to better match designs
* [ ] Tweak styling, width and positioning of columns, info, captions, etc beneath the new footer
* [ ] Update styling to ensure it works in Light mode as well as the default Dark colour scheme
* [ ] Should we add in the description to the footer (as per the designs) or leave it off for the time being?
* [ ] Should we add in a number indicator in the bottom left (e.g. 1/5)?
* [ ] Invert the fill / background colours of the icon buttons when they're selected? (It looks like this is suggested from the designs)
* [ ] Conditionally hide the info and comment icons based on the site's Jetpack settings for showing Exif metadata / comments:

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
Designs: p9Jlb4-2cI-p2

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Add an image gallery to a post on your test site
* With Jetpack active, and the Jetpack Carousel module activated, view the post on the front end of your site
* Click an image in the gallery to open up the carousel, and try out the info and comment buttons in the footer
* Go to the Jetpack settings and update the carousel to use the white colour scheme and check that the styling is correct (note: styling support for light theme hasn't been added yet) `/wp-admin/admin.php?page=jetpack#/writing`

![image](https://user-images.githubusercontent.com/14988353/122874729-a99a5e80-d376-11eb-811c-eadf6cf6750f.png)

ramonjd commented:
I've added some conditional toggling in fdf6d65305157798d03547df15ef298944f8c62f and unified things for desktop/mobile.

I might have overstepped a few things, but there's still a long way to go. 

One source of confusion for me has been the mystery right margin that is displayed when a grid row exceeds the specified height in `grid-template-rows`

<img width="583" alt="Screen Shot 2021-06-22 at 11 17 54 pm" src="https://user-images.githubusercontent.com/6458278/122931284-141bc080-d3b0-11eb-8e18-e5f2bcc1135e.png">

I think it's something to do with the hidden scroll bar?  🤷 

Safari behaves

<img width="665" alt="Screen Shot 2021-06-22 at 11 19 55 pm" src="https://user-images.githubusercontent.com/6458278/122931577-5d6c1000-d3b0-11eb-8bea-d680092b833f.png">

Will check mobile tomorrow.

I added a left margin to balance it out for now.

andrewserong commented:
@ramonjd I haven't been able to reproduce the margin issue you saw at the right hand side of the screen. Which browser were you testing with? I've removed the left margin, as it created a bit of space where you could see the previous image on the behind the footer area. I also swapped out the `span` on the icons for an `a` tag so that they can be interacted with via keyboard... but I see there's a bunch of other keyboard accessibility / tab order issues with the carousel, so maybe these can be tackled separately.

I think I'll take a look at the tap target size for the buttons next (and hover / selected state to match the designs).

andrewserong commented:
I've updated the footer icons to add a bit of nesting so that we can have a larger tap target size, but also a hover state (or an active state). Happy to revert that if it looks off, though.

I also added in styling to get this working in the light colour scheme. Not sure if the black is a bit too domineering in this scheme — we can always use a lighter grey if it's too much?

| Dark theme | Light theme |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/123026213-2254f580-d41f-11eb-8042-19eb7449bd10.png) | ![image](https://user-images.githubusercontent.com/14988353/123026222-26811300-d41f-11eb-93a1-ee4e8917bf3a.png) |

| Dark hover | Light hover |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/123026327-516b6700-d41f-11eb-9279-30bafcab4746.png) | ![image](https://user-images.githubusercontent.com/14988353/123026337-54feee00-d41f-11eb-893b-a3c0890f46a0.png) |

andrewserong commented:
Just closing out this PR in favour of #20145 which applies these changes on top of #20107 (the refactor to use Swiper.js)

ramonjd commented:
>  I haven't been able to reproduce the margin issue you saw at the right hand side of the screen. Which browser were you testing with?

@andrewserong I was just checking in Chrome on desktop, but after our rebase on top of #20107 it's gone away. Hoorah!
